File: haskell/src/Solutions/Euler348.hs
module Solutions.Euler348 where
import Utils.Numeric(isPalindrome)
import Data.List

check = group $ sort [p | a <- [2..40000], b <- [2..4000], let p = a^2 + b^3, isPalindrome p]

euler348 :: IO String
euler348 = return $ show $ sum $ map head $ filter ((==4) . length) check

File: rust/src/solutions/euler348.rs
use crate::utils::numeric::digits::Digits;
use itertools::Itertools;

pub fn euler348() -> String {
    let mut ans = 0;
    let mut palindromes = Vec::<u64>::new();
    for a in 2..=40000 {
        for b in 2..=4000 {
            let p = a * a + b * b * b;
            if p.is_palindrome() {
                palindromes.push(p);
            }
        }
    }
    palindromes.sort();
    for (p, group) in &palindromes.into_iter().group_by(|&p| p) {
        if group.count() == 4 {
            ans += p;
        }
    }
    ans.to_string()
}
